vowpalwabbit/text_parser/include/vw/text_parser/parse_example_text.h

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#pragma once
+
+#include "vw/common/string_view.h"
+#include "vw/core/multi_ex.h"
+#include "vw/core/vw_fwd.h"
+
+#include <cstdint>
+
+namespace VW
+{
+namespace parsers
+{
+namespace text
+{
+void substring_to_example(VW::workspace* all, VW::example* ae, VW::string_view example);
+
+void read_line(VW::workspace& all, example* ex, VW::string_view line);  // read example from the line.
+void read_line(VW::workspace& all, example* ex, const char* line);      // read example from the line.
+void read_lines(VW::workspace* all, const char* line, size_t len,
+    VW::multi_ex& examples);  // read examples from the new line separated strings.
+
+int read_features_string(VW::workspace* all, io_buf& buf, VW::multi_ex& examples);
+size_t read_features(io_buf& buf, char*& line, size_t& num_chars);
```

Comment:
These are probably details

Reply:
I agree for `read_features` (only used in the json parser)

I am not sure about `read_features_string`, that and `read_line` is what is set in the `parser` and used to parse, is that a detail?

Reply:
Ah I see its the "entrypoint". I guess it should not be a detail. However, it's purpose is mainly to facilitate the driver not necessary a user of the C++ API to parse input

Reply:
yup, hence the "I'm not sure what we should do with this", maybe just leave it for now until we consolidate the parser interfaces

Reply:
I think the default for when we aren't sure should be to put it in details to be honest. It at least conveys that we consider this a detail for now

Reply:
the more I think about it the more I see it as in line with `read_line`, one just takes an `io_buf` input, the other a `string_view`, and I don't think either is a detail